cloud/services/scalesets/vmss.go

```diff
@@ -339,3 +340,13 @@ func getVMSSUpdateFromVMSS(vmss compute.VirtualMachineScaleSet) (compute.Virtual
 	err = update.UnmarshalJSON(json)
 	return update, err
 }
+
+func getSecurityProfile(scaleSpec azure.ScaleSetSpec) *compute.SecurityProfile {
+	if scaleSpec.SecurityProfile == nil {
+		return nil
+	}
+
+	return &compute.SecurityProfile{
+		EncryptionAtHost: to.BoolPtr(*scaleSpec.SecurityProfile.EncryptionAtHost),
```

Comment:
same comment as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1012/files#r512541890

Reply:
Added the capability check to the function.